
This patches postfix to build on GNU/* systems. For a debian package to
build the changes to src/util/sys_defs.h, as debian/patches/20maps.dpatch
has for LINUX2, also must be done for GNU.


	Robin Elfrink <robin@15augustus.nl>



diff -uNr postfix-2.2.4.orig/makedefs postfix-2.2.4/makedefs
--- postfix-2.2.4.orig/makedefs	2005-02-22 13:35:52.000000000 +0100
+++ postfix-2.2.4/makedefs	2005-08-24 09:55:17.000000000 +0200
@@ -257,6 +257,38 @@
 		2.[0-3].*) CCARGS="$CCARGS -DNO_IPV6";;
 		esac
 		;;
+     GNU/*)	SYSTYPE=GNU
+     		# Postfix no longer needs DB 1.85 compatibility
+		if [ -f /usr/include/db.h ]
+		then
+		    : we are all set
+		    elif [ -f /usr/include/db/db.h ]
+		then
+		    CCARGS="$CCARGS -I/usr/include/db"
+		else
+		    # No, we're not going to try db1 db2 db3 etc.
+		    # On a properly installed system, Postfix builds
+		    # by including <db.h> and by linking with -ldb
+		    echo "No <db.h> include file found." 1>&2
+		    echo "Install the appropriate db*-devel package first." 1>&2
+		    echo "See the RELEASE_NOTES file for more information." 1>&2
+		    exit 1
+		fi
+		SYSLIBS="-ldb"
+		for name in nsl resolv $GDBM_LIBS
+		do
+		    for lib in /usr/lib64 /lib64 /usr/lib /lib
+		    do
+		        test -e $lib/lib$name.a -o -e $lib/lib$name.so && {
+			    SYSLIBS="$SYSLIBS -l$name"
+			    break
+			}
+		    done
+		done
+		case "$RELEASE" in
+		2.[0-3].*) CCARGS="$CCARGS -DNO_IPV6";;
+		esac
+		;;
      IRIX*.5.*)	SYSTYPE=IRIX5
 		# Use the native compiler by default
 		: ${CC=cc} ${DEBUG="-g3"}
diff -uNr postfix-2.2.4.orig/src/util/sys_defs.h postfix-2.2.4/src/util/sys_defs.h
--- postfix-2.2.4.orig/src/util/sys_defs.h	2005-06-21 23:12:00.000000000 +0200
+++ postfix-2.2.4/src/util/sys_defs.h	2005-08-24 10:00:53.000000000 +0200
@@ -686,6 +686,49 @@
 #endif
 
  /*
+  * GLibC based systems, such as GNU/Hurd or GNU/kFreeBSD. Based on LINUX2.
+  */
+#ifdef GNU
+#define SUPPORTED
+#include <sys/types.h>
+#include <features.h>
+#define USE_PATHS_H
+#define HAS_FLOCK_LOCK
+#define HAS_FCNTL_LOCK
+#define INTERNAL_LOCK   MYFLOCK_STYLE_FLOCK
+#define DEF_MAILBOX_LOCK "fcntl, dotlock"       /* RedHat >= 4.x */
+#define HAS_FSYNC
+#define HAS_DB
+#define DEF_DB_TYPE     "hash"
+#define ALIAS_DB_MAP    "hash:/etc/aliases"
+#define HAS_NIS
+#define GETTIMEOFDAY(t) gettimeofday(t,(struct timezone *) 0)
+#define ROOT_PATH       "/bin:/usr/bin:/sbin:/usr/sbin"
+#define FIONREAD_IN_TERMIOS_H
+#define USE_STATFS
+#define STATFS_IN_SYS_VFS_H
+#define UNIX_DOMAIN_CONNECT_BLOCKS_FOR_ACCEPT
+#define PREPEND_PLUS_TO_OPTSTRING
+#define HAS_POSIX_REGEXP
+#define NATIVE_SENDMAIL_PATH "/usr/sbin/sendmail"
+#define NATIVE_MAILQ_PATH "/usr/bin/mailq"
+#define NATIVE_NEWALIAS_PATH "/usr/bin/newaliases"
+#define NATIVE_COMMAND_DIR "/usr/sbin"
+#define NATIVE_DAEMON_DIR "/usr/libexec/postfix"
+#if __GLIBC__ >= 2 && __GLIBC_MINOR__ >= 1
+#define SOCKADDR_SIZE   socklen_t
+#define SOCKOPT_SIZE    socklen_t
+#endif
+#ifndef NO_IPV6
+# define HAS_IPV6
+# define HAS_PROCNET_IFINET6
+# define _PATH_PROCNET_IFINET6 "/proc/net/if_inet6"
+#endif
+#define CANT_WRITE_BEFORE_SENDING_FD
+#define HAS_DEV_URANDOM
+#endif
+
+ /*
   * HPUX11 was copied from HPUX10, but can perhaps be trimmed down a bit.
   */
 #ifdef HPUX11
